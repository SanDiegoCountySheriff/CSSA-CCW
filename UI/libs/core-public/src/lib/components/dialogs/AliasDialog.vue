<template>
  <div>
    <v-dialog v-model="dialog">
      <template #activator="{ on, attrs }">
        <v-btn
          color="primary my-5"
          v-bind="attrs"
          v-on="on"
        >
          Add Alias
        </v-btn>
      </template>

      <div class="alias-container">
        <v-form
          ref="form"
          v-model="valid"
          class="form-container"
        >
          <v-row>
            <v-col
              cols="6"
              md="5"
              sm="3"
            >
              <v-text-field
                v-model="alias.prevLastName"
                label="Previous Last Name"
                :rules="[v => !!v || 'Last name is required']"
                required
              />
            </v-col>
            <v-col
              cols="6"
              md="5"
              sm="3"
            >
              <v-text-field
                v-model="alias.prevFirstName"
                label="Previous First name"
                :rules="[v => !!v || 'First name is required']"
                required
              />
            </v-col>

            <v-col
              cols="6"
              md="5"
              sm="3"
            >
              <v-text-field
                v-model="alias.prevMiddleName"
                label="Previous Middle name"
              />
            </v-col>
          </v-row>
          <v-row>
            <v-col
              cols="6"
              md="5"
              sm="3"
            >
              <v-text-field
                v-model="alias.cityWhereChanged"
                label="City Where Changed"
              />
            </v-col>

            <v-col
              cols="6"
              md="5"
              sm="3"
            >
              <v-text-field
                v-model="alias.stateWhereChanged"
                label="State or Region where changed"
              />
            </v-col>

            <v-col
              cols="6"
              md="5"
              sm="3"
            >
              <v-text-field
                v-model="alias.courtFileNumber"
                label="Court File number"
              />
            </v-col>
          </v-row>
        </v-form>
        <div class="mt-2 btn-container">
          <v-btn
            color="success"
            @click="handleSubmit"
            class="mr-2"
          >
            Submit
          </v-btn>
          <v-btn
            color="error"
            @click="dialog = false"
          >
            Close
          </v-btn>
        </div>
      </div>
    </v-dialog>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue';
import { Alias } from '@shared-ui/types/defualtTypes';

export default defineComponent({
  name: 'AliasDialog',
  props: {
    saveAlias: {
      type: Function as PropType<(alias: Alias) => void>,
      default: () => {
        return;
      },
    },
  },
  data() {
    return {
      alias: {
        prevLastName: '',
        prevFirstName: '',
        prevMiddleName: '',
        cityWhereChanged: '',
        stateWhereChanged: '',
        courtFileNumber: '',
      } as Alias,
      dialog: false,
      valid: false,
    };
  },
  methods: {
    handleSubmit() {
      this.saveAlias(this.alias);
      this.dialog = false;
    },
  },
});
</script>

<style lang="scss" scoped>
.alias-container {
  display: flex;
  flex-direction: column;
  height: 50vh;
  width: 90%;
  justify-content: center;
  align-items: center;
  background: aliceblue;
  border-radius: 12px;
}
.btn-container {
  display: flex;
  width: 75%;
  justify-content: flex-end;
}

.form-container{
  width: 90%;
}
</style>
