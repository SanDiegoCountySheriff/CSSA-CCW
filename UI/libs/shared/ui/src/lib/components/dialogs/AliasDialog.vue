<template>
  <div>
    <v-dialog v-model="dialog">
      <template v-slot:activator="{ on, attrs }">
        <v-btn color="primary my-5" v-bind="attrs" v-on="on"> Add Alias </v-btn>
      </template>
      <div class="alias-container">
        <v-form>
          <v-row>
            <v-col>
              <v-text-field
                v-model="alias.prevLastName"
                label="Previous Last Name"
                :rules="[v => !!v || 'Last name is required']"
                required
              >
              </v-text-field>
            </v-col>
            <v-col>
              <v-text-field
                v-model="alias.prevFirstName"
                label="Previous First name"
                :rules="[v => !!v || 'First name is required']"
                required
              ></v-text-field>
            </v-col>

            <v-col>
              <v-text-field v-model="alias.prevMiddleName" label="Previous Middle name"> </v-text-field>
            </v-col>
          </v-row>
          <v-row>
            <v-col>
              <v-text-field v-model="alias.cityWhereChanged" label="City Where Changed"> </v-text-field>
            </v-col>

            <v-col>
              <v-text-field v-model="alias.stateWhereChanged" label="State or Region where changed"> </v-text-field>
            </v-col>

            <v-col>
              <v-text-field v-model="alias.courtFileNumber" label="Court File number"> </v-text-field>
            </v-col>
          </v-row>
        </v-form>
        <div class="mt-2 btn-container">
          <v-btn color="success" @click="handleSubmit" class="mr-2"> Submit </v-btn>
          <v-btn color="error" @click="dialog = false"> Close </v-btn>
        </div>
      </div>
    </v-dialog>
  </div>
</template>

<script>
export default {
  name: 'AliasDialog',
  props: {
    saveAlias: Function,
  },
  data() {
    return {
      alias: {
        prevLastName: '',
        prevFirstName: '',
        prevMiddleName: '',
        cityWhereChanged: '',
        stateWhereChanged: '',
        courtFileNumber: '',
      },
      dialog: false,
    };
  },
  methods: {
    handleSubmit() {
      this.saveAlias(this.alias);
      this.dialog = false;
    },
  },
};
</script>

<style lang="scss" scoped>
.alias-container {
  display: flex;
  flex-direction: column;
  height: 50vh;
  width: 90%;
  justify-content: center;
  align-items: center;
  background: aliceblue;
  border-radius: 12px;
}
.btn-container {
  display: flex;
  width: 75%;
  justify-content: flex-end;
}
</style>
