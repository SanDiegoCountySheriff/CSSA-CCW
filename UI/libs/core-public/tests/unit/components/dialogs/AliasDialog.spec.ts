import Vue from 'vue';
import AliasDialog from '@core-public/components/dialogs/AliasDialog.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';

Vue.use(Vuetify);
const localVue = createLocalVue();
const tMock = {
  $t: text => text,
};

describe('AliasDialog', () => {
  let vuetify;
  let wrapper;
  beforeEach(() => {
    vuetify = new Vuetify();
    //@ts-ignore
    wrapper = mount(AliasDialog, {
      localVue,
      vuetify,
      mocks: tMock,
    });
  });
  afterEach(() => {
    wrapper.destroy();
  });

  it('should render the add button', () => {
    expect(wrapper.find('button').exists()).toBeTruthy();
    expect(wrapper.find('form').exists()).toBeFalsy();
  });

  it('should render the form on button click', async () => {
    //@ts-ignore
    const onWrapper = mount(AliasDialog, {
      localVue,
      vuetify,
      mocks: tMock,
      data() {
        return {
          dialog: true,
        };
      },
    });
    expect(onWrapper.findAll('.v-input').length).toBe(6);
    onWrapper.destroy();
  });
  it('should not allow submission', async () => {
    //@ts-ignore
    const onWrapper = mount(AliasDialog, {
      localVue,
      vuetify,
      mocks: tMock,
      data() {
        return {
          dialog: true,
        };
      },
    });
    expect(onWrapper.find('#submit-btn').attributes().disabled).toBeTruthy();
    onWrapper.destroy();
  });

  it('should allow submission', async () => {
    //@ts-ignore
    const onWrapper = mount(AliasDialog, {
      localVue,
      vuetify,
      mocks: tMock,
      data() {
        return {
          dialog: true,
          valid: true,
        };
      },
    });
    expect(onWrapper.find('#submit-btn').attributes().disabled).toBeFalsy();
    onWrapper.destroy();
  });
  it('should change valid to true on inputs', async () => {
    //@ts-ignore
    const onWrapper = mount(AliasDialog, {
      localVue,
      vuetify,
      mocks: tMock,
      data() {
        return {
          dialog: true,
          valid: false,
        };
      },
    });
    onWrapper.find('#last-name').setValue('t');
    onWrapper.find('#first-name').setValue('t');
    expect(onWrapper.vm.$data.valid).toBeTruthy();
    onWrapper.destroy();
  });
});
