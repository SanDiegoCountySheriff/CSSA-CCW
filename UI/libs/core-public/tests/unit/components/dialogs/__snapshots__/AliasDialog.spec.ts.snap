// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AliasDialog should match the snapshot 1`] = `
<div data-app=""><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small primary" id="add-alias-btn" role="button" aria-haspopup="true" aria-expanded="true"><span class="v-btn__content">
        Add Alias
      </span></button>
  <div class="v-dialog__container">
    <div role="dialog" aria-modal="true" class="v-dialog__content v-dialog__content--active" style="z-index: 202;">
      <transition-stub name="dialog-transition" appear="true">
        <div tabindex="0" class="v-dialog v-dialog--active" style="transform-origin: center center;">
          <div class="alias-container" style="background: rgb(238, 238, 238);">
            <form novalidate="novalidate" class="v-form form-container">
              <div class="row">
                <div class="col-md-6 col-lg-6 col-12">
                  <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <fieldset aria-hidden="true">
                          <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                        </fieldset>
                        <div class="v-text-field__slot"><label for="last-name" class="v-label theme--light" style="left: 0px; position: absolute;">Previous Last Name</label><input maxlength="50" required="required" id="last-name" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                        <div class="v-counter theme--light">0 / 50</div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="col-md-6 col-lg-6 col-12">
                  <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <fieldset aria-hidden="true">
                          <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                        </fieldset>
                        <div class="v-text-field__slot"><label for="first-name" class="v-label theme--light" style="left: 0px; position: absolute;">Previous First name</label><input maxlength="50" required="required" id="first-name" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                        <div class="v-counter theme--light">0 / 50</div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="col-md-6 col-lg-6 col-12">
                  <div class="v-input pl-6 v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <fieldset aria-hidden="true">
                          <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                        </fieldset>
                        <div class="v-text-field__slot"><label for="input-17" class="v-label theme--light" style="left: 0px; position: absolute;">Previous Middle name</label><input maxlength="50" id="input-17" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                        <div class="v-counter theme--light">0 / 50</div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="row">
                <div class="col-md-6 col-lg-6 col-12">
                  <div class="v-input pl-6 v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <fieldset aria-hidden="true">
                          <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                        </fieldset>
                        <div class="v-text-field__slot"><label for="input-20" class="v-label theme--light" style="left: 0px; position: absolute;">City Where Changed</label><input maxlength="50" id="input-20" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                        <div class="v-counter theme--light">0 / 50</div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="col-md-6 col-lg-6 col-12">
                  <div class="v-input pl-6 v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <fieldset aria-hidden="true">
                          <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                        </fieldset>
                        <div class="v-text-field__slot"><label for="input-23" class="v-label theme--light" style="left: 0px; position: absolute;">State or Region where changed</label><input maxlength="50" id="input-23" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                        <div class="v-counter theme--light">0 / 50</div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="col-md-6 col-lg-6 col-12">
                  <div class="v-input pl-6 v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <fieldset aria-hidden="true">
                          <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                        </fieldset>
                        <div class="v-text-field__slot"><label for="input-26" class="v-label theme--light" style="left: 0px; position: absolute;">Court File number</label><input maxlength="50" id="input-26" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                        <div class="v-counter theme--light">0 / 50</div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
            <div class="mt-2 btn-container"><button type="button" class="mr-2 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small success" id="submit-btn"><span class="v-btn__content">
          Submit
        </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small error"><span class="v-btn__content">
          Close
        </span></button></div>
          </div>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;
